test/integration/tag_test.rb

```diff
@@ -42,4 +42,35 @@ def render(*)
       assert_equal(buf.object_id, output.object_id)
     end
   end
+
+  def test_tags_can_be_overwritten_using_parse_context
+    tag_name = 'testtag'
+
+    original_tag = Class.new(Block) do
+      def render(*)
+        'original_tag'
+      end
+    end
+
+    new_tag = Class.new(Block) do
+      def render(*)
+        'new_tag'
+      end
+    end
+
+    tags_overwrite = Liquid::Template::TagRegistry.new
+    tags_overwrite[tag_name] = new_tag
+
+    with_custom_tag(tag_name, original_tag) do
+      liquid = "{% #{tag_name} %} {% end#{tag_name} %}"
+
+      template = Liquid::Template.parse(liquid)
+      assert_equal('original_tag', template.render)
+
+      unless ENV['LIQUID_C'] == '1'
```

Comment:
I tried to get this working with `liquid-c`, but I wasn't comfortable enough with the codebase. We only need this for parsing and not for rendering, so selfishly there's no need to get this working in `liquid-c` for our needs. Still, if this gets merged, it would be best to have a 1:1 relationship with `liquid-c`. Open to thoughts/ideas/contributions

Reply:
I've opened https://github.com/Shopify/liquid-c/pull/181 to add liquid-c support for it.